bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/widgets/Display.java

```diff
@@ -4950,6 +4923,16 @@ void setDeviceZoom() {
 	DPIUtil.setDeviceZoom (getDeviceZoom());
 }
 
+static void cancelRootMenuTracking () {
+	long rootMenu = OS.AcquireRootMenu ();
+	if (rootMenu == 0) return; // Extra safety, not sure if it can happen
+	OS.CancelMenuTracking (rootMenu, true, 0);
+
+	// Notice from a reader:
+	// OS.ReleaseMenu() is needed here according to docs in AcquireRootMenu().
```

Comment:
Can you clarify this comment? What needs to be fixed?

Reply:
Quite literally, `OS.ReleaseMenu()` call is missing. It's needed to release the object in order to prevent the leak. I didn't want o fix it because it will need new native libs, which will prevent the patch from being cherry picked easily.

Reply:
It has always been like this, no? But does `OS.CancelMenuTracking` actually acquire the menu so it needs to be released?

Reply:
Yes, it has always been like this.
docs in `AcquireRootMenu()` say:
```
 *    The refcount of the root menu is incremented by this API. The
 *    caller should release a refcount with ReleaseMenu when it's done
 *    with the menu.
```

Reply:
But where is `AcquireRootMenu()` called?

Reply:
One line above. Check full patch diffs instead of preview.

Reply:
Aarggh!! Sorry for the noise!

Reply:
It happens :) Thanks for the testing effort you've put into this, it's been helpful!

Reply:
Thanks Alexandr! The code has been like this, so let's merge this change. But, since we may have additional calls to AcquireRootMenu() now, it would be good to fix the leak as well. Can you please create a PR to address the leak?

Reply:
Apple docs for `ReleaseMenu()` say "Deprecated: In Mac OS X 10.2 and later, use CFRelease instead of ReleaseMenu."

So could it be as simple as calling `OS.CFRelease(rootMenu);` ? I tried that and it didn't crash. ;-)

Reply:
I called `OS.objc_msgSend(rootMenu, OS.sel_retainCount);` before and after `OS.CFRelease(rootMenu);` and the retain count is decremented by 1.